### DISCUSSION TITLE
A key combination for pixel-precise moving and finest rotating.

### DISCUSSION THREAD

**Discussion by flowerhorse** [brunoherbelin/vimix#76] [Ideas]
In my work with Vimix yesterday, I had the following  case: Large gray and black rectangles were painted on a projection wall. I reproduced the rectangles with vimix in order to be able to compensate for the differences in brightness in the projection. 
The manual adjustment of the areas with the mouse without overlap and space succeeded only once. The second time, exactly one pixel width remained, which then remained visible as a line. 

If there was a simple key function e.g. SHIFT ARROW UP/DOWN/LEFT/RIGHT that would move the area only one pixel at a time, the problem would have been solved very quickly. This would also be useful for the rotate function. For artistic work with projections, this precision is needed more often.

**Comment by brunoherbelin** [comment]
Good idea to improve the precise displacement of sources. 
So, with the improvement of the Metrics of source (see #75), improved keyboard control is a good idea.

The key modifiers in vimix are:
 - CTRL + mouse : multiple selection / toggle select
 - ALT + mouse : discrete values (e.g. rotation + ALT : angles are round values)
 -  SHIFT + mouse: modified operation (e.g. scaling source becomes fixed-aspect-ratio-scaling).

The ALT + arrow keys moves sources to values at fixed steps (0.1 precision)
As per your suggestion, what I implemented in Beta is ALT + SHIFT + arrow keys to move by very tiny steps (0.001).

NB: you can also use the Metrics and enter values directly as numbers (CRTL + clic on the slider): the value is applied as you type, allowing to change it with 0.001 precision.

**Comment by brunoherbelin** [comment]
![Screenshot from 2023-04-11 00-15-14](https://user-images.githubusercontent.com/60393189/231009270-db8da245-e339-4fe1-9360-e1df6f93527d.png)

To be tested: I also changed the Metrics of Source to give position and size in pixels (inside the render area); it seems to work but I didn't check the exactitude of the coordinates. In any case a way to move and scale a source very precisely.
